core/data_center_test.go

```diff
@@ -57,112 +56,51 @@ func TestDataCenter_JSONMarshaling(t *testing.T) {
 	}
 }
 
-func TestNewDataCenterLookup(t *testing.T) {
-	type args struct {
-		idOrPermalink string
-	}
+func TestDataCenter_Ref(t *testing.T) {
 	tests := []struct {
-		name  string
-		args  args
-		want  *DataCenter
-		field FieldName
+		name string
+		obj  DataCenter
+		want DataCenterRef
 	}{
 		{
-			name:  "empty string",
-			args:  args{idOrPermalink: ""},
-			want:  &DataCenter{},
-			field: PermalinkField,
-		},
-		{
-			name:  "dc_ prefixed ID",
-			args:  args{idOrPermalink: "dc_HHwnaBCIwNHqv0aO"},
-			want:  &DataCenter{ID: "dc_HHwnaBCIwNHqv0aO"},
-			field: IDField,
-		},
-		{
-			name:  "loc_ prefixed ID",
-			args:  args{idOrPermalink: "loc_RuHTM4fyzucbYGCK"},
-			want:  &DataCenter{ID: "loc_RuHTM4fyzucbYGCK"},
-			field: IDField,
-		},
-		{
-			name:  "permalink",
-			args:  args{idOrPermalink: "country-city-1"},
-			want:  &DataCenter{Permalink: "country-city-1"},
-			field: PermalinkField,
-		},
-		{
-			name:  "random text",
-			args:  args{idOrPermalink: "dXUt33rNLmbatuAa"},
-			want:  &DataCenter{Permalink: "dXUt33rNLmbatuAa"},
-			field: PermalinkField,
+			obj:  DataCenter{ID: "dc_25d48761871e4bf"},
+			want: DataCenterRef{ID: "dc_25d48761871e4bf"},
 		},
 	}
+
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			got, field := NewDataCenterLookup(tt.args.idOrPermalink)
-
+			got := tt.obj.Ref()
 			assert.Equal(t, tt.want, got)
-			assert.Equal(t, tt.field, field)
 		})
 	}
 }
 
-func TestDataCenter_lookupReference(t *testing.T) {
+func TestDataCenterRef_queryValues(t *testing.T) {
 	tests := []struct {
 		name string
-		obj  *DataCenter
-		want *DataCenter
+		ref  DataCenterRef
 	}{
 		{
-			name: "nil",
-			obj:  nil,
-			want: nil,
+			name: "id",
+			ref:  DataCenterRef{ID: "dc_25d48761871e4bf"},
 		},
 		{
-			name: "empty",
-			obj:  &DataCenter{},
-			want: &DataCenter{},
+			name: "permalink",
+			ref:  DataCenterRef{Permalink: "central-amazon-jungle"},
 		},
 		{
-			name: "full",
-			obj: &DataCenter{
+			name: "priority",
+			ref: DataCenterRef{
 				ID:        "dc_25d48761871e4bf",
-				Name:      "Shirebury",
-				Permalink: "shirebury",
-				Country: &Country{
-					ID: "id2",
-				},
-			},
-			want: &DataCenter{ID: "dc_25d48761871e4bf"},
-		},
-		{
-			name: "no ID",
-			obj: &DataCenter{
-				Name:      "Shirebury",
-				Permalink: "shirebury",
-				Country: &Country{
-					ID: "id2",
-				},
+				Permalink: "central-amazon-jungle",
 			},
-			want: &DataCenter{Permalink: "shirebury"},
-		},
-		{
-			name: "no ID or Permalink",
-			obj: &DataCenter{
-				Name: "Shirebury",
-				Country: &Country{
-					ID: "id2",
-				},
-			},
-			want: &DataCenter{},
 		},
 	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			got := tt.obj.lookupReference()
 
-			assert.Equal(t, tt.want, got)
+	for _, tt := range tests {
+		t.Run(t.Name(), func(t *testing.T) {
```

Comment:
Small fix to use the test case name in the sub-test.

```suggestion
		t.Run(tt.name, func(t *testing.T) {
```